appgate/attrs.py

```diff
@@ -79,7 +90,57 @@ def dump_datetime(v: datetime.datetime) -> str:
     return v.isoformat(timespec="milliseconds").replace("+00:00", "Z")
 
 
-def get_dumper(platform_type: PlatformType):
+def k8s_name(name: str) -> str:
+    # This is ugly but we need to go from a bigger set of strings
+    # into a smaller one :(
+    return re.sub("[^a-z0-9-.]+", "-", name.strip().lower())
```

Comment:
This means we will need to detect conflicts at some point. Maybe add a TODO?

Reply:
yes, maybe adding some random string at the end is a good thing to do